add pub_date_sortable_string

ndushay opened the pull request.

peetucket commented on lib/stanford-mods/origin_info.rb:
```diff
@@ -38,16 +53,28 @@ def pub_date_best_single_facet_value(date_el_array)
+        return result if result
+        # settle for earliest parseable date
+        earliest = self.class.earliest_date(date_el_array)
+        return earliest[0] if earliest.present?
```
return earliest.first if earliest.present? # to match the style in other methods?

ndushay replied:
I added some comments to try to make it clearer what the data is;  it is not meant as a sortable array.  It's more of a one-element hash, really ... but I made it a 2 element array instead.  was that bad?

peetucket replied:
oh i see, i don't think its bad necessarily, but basically you are using an array to return two different values from the method, right?  you could just return (and set) two separate values in ruby, like this, if you don't want the expense of a proper data structure.  

    def some_method
      a=2
      b=3
     return a,b
    end

    x,y=some_method # sets x to 2 and y to 3
    arr=some_method # sets arr to a 2 element array of [2,3]

peetucket replied:
i should note this is one of those stylistic things that different developers would each probably do differently and i'm not sure there is a "correct" way of doing it (possibly short of declaring a custom data structure via a new class and then setting and returning that, which is overkill for this case in my opinion)

ndushay commented:
@peetucket I added some comments that hopefully clarify what is going on.  Really, earliest_date should probably return a single element hash;  instead it is returning a 2 element array.  Would it be clearer if i changed the code for it to be a single element hash?

ndushay commented:
Thanks for the code example.  I have just pushed one last tweak for improved clarity.  If it looks good to you, please hit merge.
